HeterogeneousCore/MPICore/plugins/api.cc

```diff
@@ -2,6 +2,7 @@
 #include <array>
 #include <cstring>
 #include <tuple>
+#include <iostream>
```

Comment:
Can you remove `<iostream>` now that there are is logging ?
And add `<cassert>` for `assert(...)` below ?